manifests/bases/hr/deployment.yaml

```diff
@@ -23,8 +23,8 @@ spec:
         - name: github-container-registry
       containers:
         - name: hr
-          image: ghcr.io/redbadger/hr
-          imagePullPolicy: Never
+          image: hr
+          imagePullPolicy: IfNotPresent
```

Comment:
I *think* this is what you were talking about in cell z standup?

If I do `docker build . -t ghcr.io/redbadger/hr:localtest`, it will happily say:
```
 => => writing image sha256:d1dec262af2a015125870e1f7cb8d785818b387581cd3b60acc4f3be2eef4bd9                                                         0.0s
 => => naming to ghcr.io/redbadger/hr:localtest                                                                                                                  0.0s
```

I can then `docker run ghcr.io/redbadger/hr:localtest` and it bring up the image that I just built. As long as I never run `docker push ghcr.io/redbadger/hr:localtest` (or I am never given rights to push to `ghcr.io/redbadger/hr`) then  everything stays local.

Reply:
Yep, kind of! I think my insight was that the container registry gets in the way. Everything I want is in the repo (for the sake of argument), and I want to be able to deploy it atomically, but I can't, as there are two steps. Step 1 is to deploy things to the container registry, and step 2 is to deploy to a cluster, which then fetches things from the container registry (and we have to be careful that we fetch the thing we just pushed, and not some other version of it, which is a faff, and can easily lead to errors). I think this is the problem that leads to a lot of the incidental complexity you talk about in the philosophical point above. It's probably beyond our control, but interesting to think about. 

Obviously the container registry is useful for the base images in the dockerfiles though.